src/lib/marlin_plonk_bindings/stubs/src/caml_pointer.rs

```diff
@@ -0,0 +1,49 @@
+use std::ops::{Deref, DerefMut};
+
+pub struct CamlPointer<T>(pub *mut T);
+
+impl<T> CamlPointer<T> {
+    extern "C" fn caml_pointer_finalize(v: ocaml::Value) {
+        let mut v: ocaml::Pointer<CamlPointer<T>> = ocaml::FromValue::from_value(v);
+        unsafe {
+            // Memory is freed when the variable goes out of scope
+            let _box = Box::from_raw(v.as_mut().0);
+        }
+    }
+
+    extern "C" fn caml_pointer_compare(_: ocaml::Value, _: ocaml::Value) -> i32 {
```

Comment:
Why don't we need to implement this for real?

Reply:
We should never be using polymorphic comparisons, and it never really makes sense for us to be comparing URSs, field vectors, etc. in OCaml anyway. This is only here to make the sanity checks work, so that we can roundtrip through the `deep_copy` functions and check that rust has correctly parsed and reconstructed the value using `=`.